httpapi/v2/documents_api.go

```diff
@@ -188,3 +189,125 @@ func (h handler) Commit(w http.ResponseWriter, r *http.Request) {
 	render.Status(r, http.StatusAccepted)
 	render.JSON(w, r, resp)
 }
+
+func (h handler) getDocumentWithStatus(w http.ResponseWriter, r *http.Request, st documents.Status) {
+	var err error
+	var code int
+	defer httputils.RespondIfError(&code, &err, w, r)
+
+	docID, err := hexutil.Decode(chi.URLParam(r, coreapi.DocumentIDParam))
+	if err != nil {
+		code = http.StatusBadRequest
+		log.Error(err)
+		err = coreapi.ErrInvalidDocumentID
+		return
+	}
+
+	ctx := r.Context()
+	doc, err := h.srv.GetDocument(ctx, docID, st)
+	if err != nil {
+		code = http.StatusNotFound
+		log.Error(err)
+		err = coreapi.ErrDocumentNotFound
+		return
+	}
+
+	resp, err := toDocumentResponse(doc, h.srv.tokenRegistry, jobs.NilJobID())
+	if err != nil {
+		code = http.StatusInternalServerError
+		log.Error(err)
+		return
+	}
+
+	render.Status(r, http.StatusOK)
+	render.JSON(w, r, resp)
+}
+
+// GetPendingDocument returns the pending document associated with docID.
+// @summary Returns the pending document associated with docID.
+// @description Returns the pending document associated with docID.
+// @id get_pending_document
+// @tags Documents
+// @param authorization header string true "Hex encoded centrifuge ID of the account for the intended API action"
+// @param document_id path string true "Document Identifier"
+// @produce json
+// @Failure 403 {object} httputils.HTTPError
+// @Failure 400 {object} httputils.HTTPError
+// @Failure 404 {object} httputils.HTTPError
+// @Failure 500 {object} httputils.HTTPError
+// @success 200 {object} coreapi.DocumentResponse
+// @router /v2/documents/{document_id}/pending [get]
```

Comment:
Do you think we could define an abstract getdoc endpoint? Meaning:
`@router /v2/documents/{document_id}/{status} [get]`
and add a comment as:
`@param status path string true "status" oneof pending|committed`
Or will it create collisions on other paths?

Reply:
This was done intentional because we want to allow only two kinds of the status. Having `status` of the as url path param doesn't benefit the client and makes us having more checks for status.
And this will cause collisions if we introduce a different API with similar  path.